blueocean-dashboard/src/main/js/creation/bitbucket/server/BbServerFlowManager.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { i18nTranslator } from '@jenkins-cd/blueocean-core-js';
+
+import waitAtLeast from '../../flow2/waitAtLeast';
+import BbCloudFlowManager from '../cloud/BbCloudFlowManager';
+import BbLoadingStep from '../steps/BbLoadingStep';
+
+import BbChooseServerStep from './steps/BbChooseServerStep';
+import BbServerManager from './BbServerManager';
+import STATE from './BbServerCreationState';
+
+
+const translate = i18nTranslator('blueocean-dashboard');
+const MIN_DELAY = 500;
+
+
+export default class BbServerFlowManager extends BbCloudFlowManager {
+
+    selectedServer = null;
+
+    constructor(creationApi, credentialsApi, serverApi) {
+        super(creationApi, credentialsApi);
+        this.serverManager = new BbServerManager(serverApi);
+    }
+
+    translate(key, opts) {
+        return translate(key, opts);
+    }
+
+    getStates() {
+        return STATE.values();
+    }
+
+    getState() {
+        return STATE;
+    }
+
+    getInitialStep() {
+        return {
+            stateId: STATE.PENDING_LOADING_SERVERS,
+            stepElement: <BbLoadingStep />,
+        };
+    }
+
+    onInitialized() {
+        this._loadServerList();
+        this.setPlaceholders(translate('creation.core.status.completed'));
+    }
+
+    getApiUrl() {
+        return this.selectedServer ? this.selectedServer.apiUrl : null;
+    }
+
+    _getCredentialsStepAfterStateId() {
+        return STATE.STEP_CHOOSE_SERVER;
+    }
```

Comment:
This method shows as unused in IDEA although it probably should be called from `_findExistingCredentialsComplete` in the "Cloud" manager - see my comment above. You probably need to add this method to the "Cloud" manager and return value of `null` since the step can be added at the beginning every time.